crates/bevy_ecs/src/observer/runner.rs

```diff
@@ -296,6 +297,33 @@ impl Observer {
     pub fn descriptor(&self) -> &ObserverDescriptor {
         &self.descriptor
     }
+
+    /// Creates a [`Bundle`] that will spawn a new [`Observer`] (its own Entity)
+    /// observing this entity it's spawned on.
+    ///
+    /// Example
+    /// ```rust
+    ///     commands.spawn(
+    ///         Node::default(),
+    ///         Text::from("Not clicked"),
+    ///         Observer::as_bundle(
+    ///             |trigger: Trigger<Pointer<Click>>,
+    ///              mut text: Query<&mut Text>| {
+    ///                **text.get_mut(trigger.target).unwrap() = "Clicked!".to_string();
+    ///          })
+    ///     )
+    /// ```
+    pub fn as_bundle<E, B, M, O>(observer: O) -> impl Bundle
```

Comment:
I don't like this name: it leaks too many implementation details and implies that the observer component is added to the same entity. Something like `Observe::watch_this` (uh, please bikeshed) would much more clearly communicate the intent in this code snippet.